pkg/infrared/infrared.go

```diff
@@ -234,3 +260,39 @@ func (ir *Infrared) copy(dst io.WriteCloser, src io.ReadCloser, srcClosedChan ch
 	io.CopyBuffer(dst, src, *b)
 	srcClosedChan <- struct{}{}
 }
+
+func (ir *Infrared) addConn(c *conn) {
+	ir.mu.Lock()
+	defer ir.mu.Unlock()
+	ir.conns[c.RemoteAddr()] = c
+}
+
+func (ir *Infrared) removeConn(c *conn) {
+	ir.mu.Lock()
+	defer ir.mu.Unlock()
+	delete(ir.conns, c.RemoteAddr())
+}
+
+func writeProxyProtocolHeader(addr net.Addr, rc net.Conn) error {
+	rcAddr := rc.RemoteAddr()
+	tcpAddr := rcAddr.(*net.TCPAddr)
+
+	tp := proxyproto.TCPv4
+	if tcpAddr.IP.To4() == nil {
+		tp = proxyproto.TCPv6
+	}
+
+	header := &proxyproto.Header{
+		Version:           2,
+		Command:           proxyproto.PROXY,
+		TransportProtocol: tp,
+		SourceAddr:        addr,
+		DestinationAddr:   rcAddr,
+	}
+
+	if _, err := header.WriteTo(rc); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
```suggestion
	_, err := header.WriteTo(rc)

	return err
```

you can also just do this, because if WriteTo doesn't return an error, err will be nil as well

Reply:
Yes, but I like it as it is right now. Even if it doesn't do anything. Not sure what the compiler does under the hood though